packages/jira-adapter/src/filters/assets/assets_object_field_configuration_references.ts

```diff
@@ -0,0 +1,129 @@
+  attributesDisplayedOnIssue: Joi.array().items(Joi.alternatives(Joi.object(), Joi.string())),
+}).unknown(true)
+
+const isAssetsObjectFieldConfiguration = createSchemeGuard<assetObjectFieldConfiguration>(
+  ASSETS_OBJECT_FIELD_CONFIGURATION_SCHEME,
+  'Received an invalid assets object field configuration',
+)
+
+const createObjectTypeAttributeReferences = (
+  assetsObjectFieldConfiguration: assetObjectFieldConfiguration,
+  objectSchemaToAttributesByName: Record<string, Record<string, InstanceElement>>,
+  field: 'attributesIncludedInAutoCompleteSearch' | 'attributesDisplayedOnIssue',
+): (ReferenceExpression | string)[] =>
+  makeArray(assetsObjectFieldConfiguration[field])
+    .filter(_.isString)
+    .map(attributeName => {
+      const objectSchemaFullName = assetsObjectFieldConfiguration.objectSchemaId.elemID.getFullName()
+      const attributeInstance = objectSchemaToAttributesByName[objectSchemaFullName]?.[attributeName]
+      if (isInstanceElement(attributeInstance)) {
+        return new ReferenceExpression(attributeInstance.elemID, attributeInstance)
+      }
+      return attributeName
+    })
+    .filter(isDefined)
+
+/* 
+This filter is responsible for creating references from the field context of `assetsObjectField`
+to `ObjectTypeAttribute` instances. 
+We create these references here because the uniqueness of ObjectTypeAttribute is determined by a combination of objectSchema, objectType and the attribute name.
+We can describe the relation between the elements as follows:
+ObjectSchema -> ObjectType[] -> ObjectTypeAttribute[]
+ObjectTypeAttribute has a field called objectType, which is a reference to an ObjectType.
+ObjectType's parent is a reference to ObjectSchema.
+*/
+const filter: FilterCreator = ({ config }) => ({
+  name: 'assetsObjectFieldConfigurationReferencesFilter',
+  onFetch: async elements => {
+    if (!config.fetch.enableAssetsObjectFieldConfiguration) {
+      return
+    }
+
+    const objectTypeAttributes = elements
+      .filter(isInstanceElement)
+      .filter(instance => instance.elemID.typeName === OBJECT_TYPE_ATTRIBUTE_TYPE)
+      .filter(instance => isReferenceExpression(instance.value.objectType))
+
+    const objectTypeAttributesByObjectType = _.groupBy(objectTypeAttributes, attribute =>
+      attribute.value.objectType.elemID.getFullName(),
+    )
+
+    const objectSchemaToObjectTypes = _.groupBy(
+      elements.filter(isInstanceElement).filter(instance => instance.elemID.typeName === OBJECT_TYPE_TYPE),
+      objectType => getParent(objectType).elemID.getFullName(),
+    )
+
+    const objectSchemaToAttributesByName = _.mapValues(objectSchemaToObjectTypes, objectTypes =>
+      _.keyBy(
+        objectTypes
+          .flatMap(objectType => objectTypeAttributesByObjectType[objectType.elemID.getFullName()])
+          .filter(isInstanceElement),
+        attributeInstance => attributeInstance.value.name as string,
+      ),
+    )
```

Comment:
At this point each objectTypeSchema has all the attributes beneath it in the attributes field. I think its much more cleaner to use this at this point (And they are already references)

```suggestion
    const objectSchemaToAttributesReferenceByName = Object.fromEntries(elements
    .filter(isInstanceElement)
    .filter(instance => instance.elemID.typeName === OBJECT_SCHEMA_TYPE)
    .map(instance => [
      instance.elemID.getFullName(),
      _.keyBy(instance.value.attributes, ref => ref.value.value.name), // Transform attributes array to a record keyed by 'name'
    ]))
```

Reply:
I did not know that, thanks!